### PR TITLE
fix: upstream.protocol description for config v3

### PR DESCRIPTION
**Pull request by nijikokun**
Improves the description and updates the table to include the correct values and missing description.

### CODE REVIEW

File: docs/agent/config/v3.mdx [hunk 1]
```diff
@@ -344,17 +344,17 @@ ngrok start --all
...
+| [`traffic_policy`](#traffic-policy)                  |          | Accepts a Traffic Policy configuration in YAML format.                                                                                                                     |
+| [`upstream`](#upstream)                              | Required | Upstream service configuration options.                                                                                                                                    |
+| [`upstream.url`](#upstreamurl)                       | Required | The address you would like to forward traffic to.                                                                                                                          |
+| [`upstream.protocol`](#upstreamprotocol)             |          | The application layer protocol the agent should use when talking to your upstream application for HTTP endpoints.                                                          |
```
Comment (tammybailey) [inline docs/agent/config/v3.mdx:356]: Suggest adding something like "Accepted values are `http1` (default) and `http2`" here.

File: docs/agent/config/v3.mdx [hunk 2]
```diff
@@ -461,11 +461,11 @@ endpoints:
...
+| Name                                                 | Required | Description                                                                                                                                                                 |
+| ---------------------------------------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| [`upstream.url`](#upstreamurl)                       | Required | The address you would like to forward traffic to.                                                                                                                           |
+| [`upstream.protocol`](#upstreamprotocol)             |          | The application layer protocol the agent should use when talking to your upstream application for HTTP endpoints.                                                           |
```
Comment (tammybailey) [inline docs/agent/config/v3.mdx:467]: Suggest adding "Accepted values are `http1` (default) and `http2`" here.

**Review by tammybailey — APPROVED** [review]
lgtm with a few suggestions

File: docs/agent/config/v3.mdx (hunk 2 above)
Comment (nijikokun) [inline docs/agent/config/v3.mdx:467]:
```suggestion
| [`upstream.protocol`](#upstreamprotocol)             |          | The application layer protocol the agent should use when talking to your upstream application for HTTP endpoints. Accepted values are `http1` (default) and `http2`.                                                          |
```

File: docs/agent/config/v3.mdx (hunk 1 above)
Comment (nijikokun) [inline docs/agent/config/v3.mdx:356]:
```suggestion
| [`upstream.protocol`](#upstreamprotocol)             |          | The application layer protocol the agent should use when talking to your upstream application for HTTP endpoints.  Accepted values are `http1` (default) and `http2`.                                                         |
```

**Pull request merged**